### PR TITLE
Improve scaffolder docs

### PR DESCRIPTION
**Pull request by sblausten**
https://app.shortcut.com/larder/story/23952/uplight-extending-existing-templates

https://deploy-preview-1444--roadie.netlify.app/docs/scaffolder/writing-templates/

- Some folder changes to make navigating our code easier
- Fix a few things and improve page on grouping and certified templates which was orphaned from nav
- Add examples on placeholder usage with external files 
- Add example of placeholder usage for steps

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,6 +67,7 @@
     "react-tabs": "^4.2.1",
     "react-use": "^17.2.4",
     "remark-html": "^13.0.2",
+    "sharp": "^0.33.5",
```
Comment (sblausten) [inline package.json:70]: This seemed to be needed when I build it locally

File: content/docs/docs-nav.yaml [hunk 2]
```diff
@@ -125,24 +125,25 @@ nav:
           - Actions Directory: '/docs/scaffolder/scaffolder-actions-directory/'
           - Custom Scaffolder Actions: '/docs/scaffolder/self-hosted-scaffolder-actions/'
       - Scaffolder Parameters:
-          - Paremeters Guide: '/docs/scaffolder/scaffolder-parameters'
+          - Paremeters Guide: '/docs/scaffolder/scaffolder-parameters/'
           - Custom Scaffolder Parameters: '/docs/scaffolder/custom-fields/'
+      - Organising Templates: '/docs/scaffolder/certified-templates/'
```
Comment (sblausten) [inline content/docs/docs-nav.yaml:130]: This was missing from nav

File: content/docs/docs-nav.yaml [hunk 3]
```diff
@@ -121,28 +121,28 @@ nav:
           - Publishing Plugins: '/docs/custom-plugins/legacy/artifactory/'
   - Scaffolder:
       - Overview: '/docs/scaffolder/writing-templates/'
-      - Scaffolder Actions:
-          - Actions Directory: '/docs/scaffolder/scaffolder-actions-directory/'
-          - Custom Scaffolder Actions: '/docs/scaffolder/self-hosted-scaffolder-actions/'
-      - Scaffolder Parameters:
-          - Paremeters Guide: '/docs/scaffolder/scaffolder-parameters'
-          - Custom Scaffolder Parameters: '/docs/scaffolder/custom-fields/'
+      - Actions Directory: '/docs/scaffolder/scaffolder-actions-directory/'
```
Comment (sblausten) [inline content/docs/docs-nav.yaml:124]: Flattening these as they are not deep enough to warrent being hidden

File: content/docs/docs-nav.yaml [hunk 4]
```diff
@@ -121,28 +121,28 @@ nav:
...
-          - Transform YAML files with JSONata: '/docs/scaffolder/yaml-jsonata-transform'
-          - Using for loops in templates: '/docs/scaffolder/loops-in-templates'
-          - Using for loops to repeat scaffolder steps: '/docs/scaffolder/loops-in-steps'
-          - Assigning variables to the scaffolder context: '/docs/scaffolder/variables-in-scaffolder'
-          - Creating a PR to an Azure DevOps repository: '/docs/scaffolder/azure-actions'
+          - Create RFC: '/docs/scaffolder/recipes/create-rfc/'
+          - Debug template: '/docs/scaffolder/recipes/debug-template/'
+          - Append file: '/docs/scaffolder/recipes/append/'
+          - Re-use parameters: '/docs/scaffolder/recipes/global-constants/'
+          - Re-use steps: '/docs/scaffolder/recipes/placeholders/'
```
Comment (sblausten) [inline content/docs/docs-nav.yaml:134]: Adding this

File: content/docs/scaffolder/certified-templates/index.md [hunk 5]
```diff
@@ -1,12 +1,25 @@
 ---
-title: Certified Templates
+title: Organising Templates
```
Comment (sblausten) [inline content/docs/scaffolder/certified-templates/index.md:2]: Repurposing this for a more general page

**Review by Xantier — APPROVED** [review]

**Pull request merged**